bmwqemu.pm

```diff
@@ -180,7 +181,12 @@ sub log_format_callback {
     # ensure indentation for multi-line output
     $lines =~ s/(?<!\A)^/  /gm;
 
-    return '[' . Time::Moment->now . "] [$level] " . $lines;
+    my $idx = index($0, ': ');
+    my $proc = sprintf('%.10s[%d]', $idx != -1 ? substr($0, $idx + 2) : basename($0), $$);
+
+    return sprintf('%s %s [%s] %s',
+        Time::Moment->now_utc()->with_precision(3)->to_string(),
+        $proc, $level, $lines);
```

Comment:
I just recently consolidated on the time format with other tooling, e.g. openQA, and simplified *and* optimized runtime-overhead with the use of Time::Moment. Going back to a custom sprintf for every single line has a non-significant performance impact which we should avoid

Reply:
Actually, for some parts (of openQA) the logging overhead was notable. I'd also prefer to keep the time formatting as-is.